lib/gears/table.lua

```diff
@@ -212,5 +212,20 @@ function gtable.merge(t, set)
     return t
 end
 
+--- Map a function to a table. The function is applied to each value
+-- on the table, returning a table of the results.
+-- @class function
+-- @name map
+-- @tparam function f the function to be applied to each value on the table
+-- @tparam table tbl the container table whose values will be operated on
+-- @treturn table Return a table of the results
+function gtable.map(f, tbl)
```

Comment:
In defense of the argument order, this is traditionally the way the map function arguments are in all languages.
Python: `map(function_to_apply, list_of_inputs)`
Haskell: `map :: (a -> b) -> [a] -> [b]`
PHP: `array array_map ( callable $callback , array $array1 [, array $... ] )`
Even the [Lua example](http://lua-users.org/wiki/FunctionalLibrary) has `(func, tbl)`